include/seqan3/range/view/take_until.hpp

```diff
@@ -235,17 +235,18 @@ class view_take_until : public ranges::view_interface<view_take_until<urng_t, fu
         //!\brief Defaulted.
         ~iterator_type_consume_input()                                                             = default;
 
-        //!\brief Constructor that delegates to the CRTP layer.
-        iterator_type_consume_input(base_base_t it) noexcept(noexcept(base_t{it})) :
-            base_t{std::move(it)}
-        {}
-
         //!\brief Constructor that delegates to the CRTP layer and initialises the callable.
         iterator_type_consume_input(base_base_t it,
                                     fun_ref_t _fun,
                                     sentinel_type sen) noexcept(noexcept(base_t{it})) :
             base_t{std::move(it)}, fun{_fun}, stored_end{std::move(sen)}
-        {}
+        {
+            while (fun(**this))
+            {
+                at_end_gracefully = true;
+                base_t::operator++();
+            }
+        }
```

Comment:
This invokes UB on multiple ocasions, because you never check at end. I would recommend doing
```cpp
if ((*this->this_to_base() != stored_end) && fun(**this))
    ++(*this);
```
The increment operator will take care of the rest.

Reply:
this is not completely correct because you skip the first character that evaluated to true and `at_end_gracefully` was not set :o

Reply:
any preference?

Reply:
either is fine